aspnetcore/fundamentals/primitives/change-tokens.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Detect changes with Change Tokens in ASP.NET Core
+author: guardrex
+description: Learn how to use Change Tokens to track changes to configuration, arbitrary files, and objects in apps.
+ms.author: riande
+manager: wpickett
+ms.date: 11/06/2017
+ms.topic: article
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: fundamentals/primitives/change-tokens
+---
+# Detect changes with Change Tokens in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+[ChangeToken](/dotnet/api/microsoft.extensions.primitives.changetoken) is a general-purpose, low-level building block used to track changes.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/primitives/change-tokens/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## ChangeToken class
+
+`ChangeToken` is a static class used to propagate notifications that a change has occurred. `ChangeToken` resides in the [Microsoft.Extensions.Primitives](/dotnet/api/microsoft.extensions.primitives) namespace. For apps that don't use the [Microsoft.AspNetCore.All](https://www.nuget.org/packages/Microsoft.AspNetCore.All/) metapackage, reference the [Microsoft.Extensions.Primitives](https://www.nuget.org/packages/Microsoft.Extensions.Primitives/) NuGet package in the project file.
+
+`ChangeToken`'s [OnChange(Func\<IChangeToken>, Action)](/dotnet/api/microsoft.extensions.primitives.changetoken.onchange?view=aspnetcore-2.0#Microsoft_Extensions_Primitives_ChangeToken_OnChange_System_Func_Microsoft_Extensions_Primitives_IChangeToken__System_Action_) method registers an `Action` to call whenever the token changes:
+* `Func<IChangeToken>` produces the token.
+* `Action` is called when the token changes.
+
+`ChangeToken`'s [OnChange\<TState>(Func\<IChangeToken>, Action\<TState>, TState)](/dotnet/api/microsoft.extensions.primitives.changetoken.onchange?view=aspnetcore-2.0#Microsoft_Extensions_Primitives_ChangeToken_OnChange__1_System_Func_Microsoft_Extensions_Primitives_IChangeToken__System_Action___0____0_) method registers an `Action` to call whenever the token changes with state tracking:
+* `TState` is state for the token consumer `Action`.
+* `Func<IChangeToken>` produces the token.
+* `Action` is called when the token changes.
+
+## IChangeToken interface
```

Comment:
Ordering? Move IChangeToken ahead of ChangeToken? IChangeToken is far more generally applicable.